codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -196,8 +196,8 @@ protected ByteBuf internalBuffer() {
     @Override
     public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
         ByteBuf buf = internalBuffer();
-        int readable = buf.readableBytes();
         if (buf.isReadable()) {
+            int readable = buf.readableBytes();
```

Comment:
Actually even better would be to do:

```java
int readable = buf.readableBytes();
if (readable > 0) {
    ....
}
```

This way you have only one method call in both cases. Would you please update the PR with this and ping me again ?